### PR TITLE
Feat: Delete account API client

### PR DESCRIPTION
**Pull request by kanta1207**
## Overview

Implement "Delete account" API client

## Changes

- Add `deleteUser` API client
- Add `removeUser` action
- Apply `removeUser` and `signOut` to `DeleteAccountDialog`

## Screen Captures

https://github.com/nishiki-tech/nishiki-frontend/assets/99339182/8efc3911-68fc-4f1e-a9bf-a3c21e21d2b6

## Assignee Checklist:

- [x] The naming convention of the PR title is correct (See the comment at the top of this template)
- [x] The base branch is correct (See: [Types of Branches](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#types-of-branches))
- [x] The branch name follows the [Branch Naming Conventions](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#branch-naming-conventions)
- [x] The correct assignees and reviewers have been designated for this PR
- [x] The coding style follows the [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [x] All the related issues are associated with this PR
- [x] All criteria in the associated issues are met (please tick the checkboxes)
- [x] My changes do not generate new warnings or errors (especially in the console)

## Reviewer Checklist:

- [ ] Double-check the "Assignee Checklist"
- [x] The code follows the generic best practices and our [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [x] The code is well-commented and easy to understand
- [x] The UI changes accurately reflect the provided design

### CODE REVIEW

File: src/components/pages/ProfilePage.tsx [hunk 1]
```diff
@@ -31,7 +31,10 @@ export const ProfilePage = async () => {
   const { userId, name } = await getUserInfo();
 
   return (
-    <MobileLayout heading="Profile" headerRight={<ProfileHeaderDropdownMenuTriggerButton />}>
+    <MobileLayout
+      heading="Profile"
+      headerRight={<ProfileHeaderDropdownMenuTriggerButton userId={userId} />}
```
Comment (kanta1207) [inline src/components/pages/ProfilePage.tsx:36]: Props drilling of `userId` starts from here, maybe better to refactor it in the future by using global state for example.

File: src/features/profile/components/ProfileHeaderDropdownMenuTriggerButton/ProfileHeaderDropdownMenuContent/DeleteAccountDialogTriggerButton/DeleteAccountDialogContent.tsx [hunk 2]
```diff
@@ -35,7 +45,17 @@ export const DeleteAccountDialogContent = ({ onParentClose }: IDeleteAccountDial
    * @returns void
    */
   const handleDelete = async () => {
-    alert('delete account');
+    const result = await removeUser(userId);
+    if (!result.ok) {
+      alert('Something went wrong. Please try again.');
+    } else {
+      /**
```
Comment (kanta1207) [inline src/features/profile/components/ProfileHeaderDropdownMenuTriggerButton/ProfileHeaderDropdownMenuContent/DeleteAccountDialogTriggerButton/DeleteAccountDialogContent.tsx:52]: Without `signOut(()` in here, invalid auth token will remain in the client

**Review by nick-y-ito — APPROVED** [review]
@kanta1207 
Thank you for your work! I couldn't find any problems with your implementation!

**Comment by kanta1207** [comment]
@nick-y-ito 
Thank you sooo much for your review! 🙌

**Pull request merged**